BUILD.md

```diff
@@ -0,0 +1,88 @@
+# Building DLAF
+
+## Build using Spack
+
+Example installation:
+
+`spack install dla-future ^[virtuals=blas,lapack] intel-oneapi-mkl`
+
+or you can also build C API compatible with ScaLAPACK with:
+
+`spack install dla-future +scalapack ^[virtuals=blas,lapack,scalapack] intel-oneapi-mkl`
+
+You can go even further with a more detailed spec like this one, which builds dla-future in debug mode, using the clang compiler, specifying that the pika on which it depends has to be built
+in debug mode too, and that we want to use MPICH as MPI implementation, without fortran support (because clang does not support it).
```

Comment:
```suggestion
in debug mode too, and that we want to use MPICH as MPI implementation, without Fortran support (because clang does not support it).
```